resources/fetching/fetchers/azure/assets_enricher_storage_account.go

```diff
@@ -92,12 +97,56 @@ func (e storageAccountEnricher) addStorageAccountBlobServices(ctx context.Contex
 		return err
 	}
 
-	if len(dataProtection) == 0 {
+	return e.appendExtensionAssets(inventory.ExtensionBlobService, dataProtection, assets)
+}
+
+func (e storageAccountEnricher) addStorageAccountServicesDiagnosticSettings(ctx context.Context, storageAccounts []inventory.AzureAsset, assets []inventory.AzureAsset) error {
+	if len(storageAccounts) == 0 {
+		return nil
+	}
+
+	if err := e.addBlobDiagnosticSettings(ctx, storageAccounts, assets); err != nil {
+		return err
```

Comment:
if the blob diagnostics fails don't we want to "soft fail" and keep on retrieving the other diagnostic settings? I mean, one doesn't mean all will fail

Reply:
Yes, in general, it might be better for the enricher to be best-effort. 

But I would open a small separate ticket to change the previous parts as well.